### PR TITLE
Manage groups on an integration

### PR DESCRIPTION
**Pull request by JedrzejJanasiak**
## Issue

https://gravitee.atlassian.net/browse/APIM-6136

## Description

Implementation of group management for Integrations with JDBC and MongoDB support. 

## Additional context

PR also contains a small fix for the Primary Owner migration script run on APIM with the JDBC database. 

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-knlntdzysl.chromatic.com)

### CODE REVIEW

File: gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/resources/liquibase/changelogs/v4_5_0/05_add_integration_groups_table.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+databaseChangeLog:
+    - changeSet:
+          id: 4.5.0_add_integration_groups
+          author: GraviteeSource Team
+          changes:
+              - createTable:
+                    tableName: ${gravitee_prefix}integrations_groups
```
Comment (JedrzejJanasiak) [inline gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/resources/liquibase/changelogs/v4_5_0/05_add_integration_groups_table.yml]: I'm not so sure about this naming 🤔 @jgiovaresco @michel-barret wdyt?

Comment (jgiovaresco) [reply gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/resources/liquibase/changelogs/v4_5_0/05_add_integration_groups_table.yml]: If we follow what is done for APIs and Applications it should be `${gravitee_prefix}integration_groups`

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/openapi-apis.yaml [hunk 2]
```diff
@@ -3694,8 +3694,15 @@ components:
                 pendingJob:
                     $ref: "#/components/schemas/IngestionJob"
                 primaryOwner:
-                  $ref: "#/components/schemas/PrimaryOwner"
-
+                    $ref: "#/components/schemas/PrimaryOwner"
+                groups:
+                    type: array
+                    description: Integration's groups
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/openapi-apis.yaml]:
```suggestion
                    description: A list of group ids
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/openapi-apis.yaml [hunk 3]
```diff
@@ -3694,8 +3694,15 @@ components:
...
+                    example:
+                        - MY_GROUP1
+                        - MY_GROUP2
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/openapi-apis.yaml]:
```suggestion
                        - 3dc1bc32-1411-4262-b3a2-058222569670
                        - 63cd7036-694f-4c40-833f-84c4a5d39544
```

**Review by michel-barret — APPROVED** [review]

**Review by jgiovaresco — APPROVED** [review]

**Pull request merged**